### ISSUE TITLE
Subword regularization

### ISSUE THREAD

**Issue by nikitakit** [huggingface/tokenizers#730]
As far as I can tell, this library doesn't currently implement [subword regularization](https://arxiv.org/abs/1804.10959).

I've found BPE dropout to be quite helpful when using BPE-based models like GPT-2, and I'd like to have the option to try subword regularization for Unigram models (like T5) as well.

I should note that I'm using pre-trained models via the `transformers` package, and my code requires offset mappings for aligning subword tokens with characters in the input string. Even though the so-called "slow" tokenizers support subword regularization, their lack of offset mapping capabilities is a non-starter for my application.

**Comment by Narsil** [comment]
Hi, 

`sample` algorithm is indeed not implemented, it should be however relatively simple to do (just look at sentencepiece code, we have a very similar structure here), PRs welcome.
